src/rust/ensogl/lib/text/src/buffer/view.rs

```diff
@@ -610,19 +638,23 @@ impl ViewBuffer {
     ///
     /// It returns selection after modification and byte offset of the next selection ranges.
     fn modify_selection
-    (&self, selection:Selection, text:&Text, transform:Option<Transform>) -> (Selection,Bytes) {
+    (&self, selection:Selection, text:Text, transform:Option<Transform>) -> Modification {
```

Comment:
why the text is not passed as reference anymore?

Reply:
Because it is returned in Modification, so it would be cloned anyway. And it should not be a big problem, as Text implements Copy-on-write.